src/read.c

```diff
@@ -3535,18 +3540,10 @@ static avifResult avifParseCondensedImageBox(avifMeta * meta, uint64_t rawOffset
     // The ExtendedMetaBox may reuse items and properties created above so it must be parsed last.
 
     if (hasExtendedMeta) {
+        assert(avifROStreamOffset(&s) + extendedMetaSize <= rawLen);
```

Comment:
Similarly, avoid using `rawLen` (the size of stream `s`) expliclty and use a suitable avifROStream function instead. We can say `assert(avifROStreamHasBytesLeft(&s, extendedMetaSize);`.